usr/src/uts/armv8/os/ddi_impl.c

```diff
@@ -3083,40 +3082,44 @@ i_ddi_cacheattr_to_hatacc(uint_t flags, uint_t *hataccp)
 }
 
 static int
-get_address_cells(pnode_t node)
+get_address_cells(dev_info_t *dip)
 {
-	int address_cells = 0;
+	int address_cells = 2;
 
-	while (node > 0) {
-		int len = prom_getproplen(node, "#address-cells");
-		if (len > 0) {
-			ASSERT(len == sizeof (int));
-			int prop;
-			prom_getprop(node, "#address-cells", (caddr_t)&prop);
-			address_cells = ntohl(prop);
+	for (;;) {
```

Comment:
This loop seems to just be emulating the effect of not passing `DDI_PROP_DONTPASS`.  I think you could just remove this whole function, and just `ddi_prop_get_int()` where we need it

Reply:
There's only one difference here - when we elide `DDI_PROP_DONTPASS` and the search gets up to the root node _and_ not results are found there, then the options code is consulted. If that _node_ does _not_ exist we panic (in a DEBUG build).

I'm happy to change this though - I've never liked these functions.